action.yaml

```diff
@@ -182,12 +182,18 @@ runs:
         fi
 
     - name: Setup Java
+      id: setup-java
       if: steps.detect-java.outputs.VERSION != ''
       uses: actions/setup-java@v4
       with:
         distribution: ${{ steps.detect-java.outputs.DISTRIBUTION }}
         java-version: ${{ steps.detect-java.outputs.VERSION }}
 
+    - name: Export JAVA_HOME for subsequent steps
+      shell: bash
+      if: steps.detect-java.outputs.VERSION != ''
+      run: echo "JAVA_HOME=${{ steps.setup-java.outputs.path }}" >> $GITHUB_ENV
```

Comment:
Is there a test you could add to reproduce the issue you were having on the other repo to avoid it comes back in the future